operator/cmd/mesh/manifest-apply.go

```diff
@@ -213,3 +225,32 @@ func ApplyManifests(setOverlay []string, inFilenames []string, force bool, dryRu
 
 	return nil
 }
+
+// DetectIstioVersionDiff will show warning if istioctl version and control plane version are different
+// nolint: interfacer
+func DetectIstioVersionDiff(cmd *cobra.Command, obv string, maArgs *manifestApplyArgs) error {
+	var opts clioptions.ControlPlaneOptions
+	icps, err := GetVersionInfo(maArgs.kubeConfigPath, maArgs.context, opts)
+	if err != nil {
+		return err
+	}
+	if len(*icps) != 0 {
+		var icpVer string
+		for _, icp := range *icps {
+			icpVer = icp.Info.Version
```

Comment:
This won't work in the case there are different versions

Reply:
Do we want to compare it with all the control plan version? 
OR
Only one different version will be enough to display warning?

Reply:
Is it ok to use `[0].version` for comparison?

eg: [retrieveControlPlaneVersion](https://github.com/istio/istio/blob/master/operator/cmd/mesh/upgrade.go#L340) and [waitUpgradeComplete](https://github.com/istio/istio/blob/master/operator/cmd/mesh/upgrade.go#L345) uses it.

Reply:
This will consider if at least one different control plan version is installed and will display a warning.
@howardjohn PTAL ?